### PR TITLE
kovri-install.sh: moved catch(). Fixes #876

### LINKED ISSUE #876: kovri-install.sh: line 117: catch: command not found
---
**By submitting this issue, I confirm the following:**

- I have read and understood the contributor guide in [kovri-docs](https://github.com/monero-project/kovri-docs).
- I have checked that the issue I am reporting can be replicated or that the feature I am suggesting is not present.
- I have checked opened or recently closed [pull requests](https://github.com/monero-project/kovri/pulls) for existing solutions/implementations to my issue/suggestion.
---

> make -j8 DESTDIR=/var/tmp/portage/net-vpn/kovri-9999/image/ install
./pkg/installers/**kovri-install.sh: line 117: catch: command not found**
The Kovri I2P Router Project (c) 2015-2017

The catch() function in pkg/installers/kovri-install.sh is defined after it's first used outside a function. This leads to the above error.

### PR DESCRIPTION
**Pull request by Wuodan**
---
**By submitting this pull-request, I confirm the following:**

- I have read and understood the contributor guide in [kovri-docs](https://github.com/monero-project/kovri-docs).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.
---

### CODE REVIEW

File: pkg/installers/kovri-install.sh [hunk 1]
```diff
@@ -107,7 +117,7 @@ shift "$(($OPTIND -1))"
 
 # Set default resources if needed
 if [[ -z $resources ]]; then
-  resources="pkg/client pkg/config contrib/utils/kovri-bash.sh build/${bins[0]} build/${bins[1]}"
+  resources="pkg/client pkg/config contrib/utils/kovri-bash.sh build/${bins[0]}"
```
Comment (anonimal) [inline pkg/installers/kovri-install.sh]: This isn't optional though as it's part of our release build.

Comment (anonimal) [reply pkg/installers/kovri-install.sh]: Could you instead apply a workaround for non-release installs?

Comment (Wuodan) [reply pkg/installers/kovri-install.sh]: Sorry, I didn't mean for that second commit to be in this PR, only the mini issue with the catch() function.

Will propose a patch for the Makefile for non-release installs to build kovri-util instead.

I'm using Gentoo Linux amd64 and arm, want to package kovri.
Installing in Gentoo by default means to build from source, so the package-management is not much more then compile and install scripts in a chroot. Shell is bash.

Comment (Wuodan) [reply pkg/installers/kovri-install.sh]: About kovri-util being optional or not:
In the Makefile, it's not used unless for release-builds.
The proposed change to the installer still installs kovri-util, it just does not check for its presence as required resource (which triggers the catch() function).

Should I create another PR where the Makefile builds kovri-util for non-release installs? Thus making kovri-util no longer 'optional'?

**Comment by anonimal** [comment]
You didn't need to close this PR. You could've kept adding to it instead of opening #880.

**Comment by anonimal** [comment]
JFTR: the catch issue is not present in bash 4.3.48 (Ubuntu) but is present with 4.4.19 (Arch). I imagine gentoo is using the 4.4 series.

**Comment by Wuodan** [comment]
bash, version 4.4.12, correct. Sorry for closing the PR, it automatically with rebase, force push and new branch.

**Pull request closed without merging**